### PR TITLE
Display boost multiplier on StakingDialog modal

### PR DESCRIPTION
**Pull request by nguyenaiden**
***Description***: On the staking dialog modal, user's current boost multiplier will be calculated and displayed. If the user hasn't provided LP or staked any asset, the multiplier will show up as `-`
image
image

### CODE REVIEW

File: src/hooks/useUserGauge.ts [hunk 1]
```diff
@@ -54,6 +57,21 @@ export default function useUserGauge(gaugeAddress?: string): UserGauge | null {
     userGaugeRewards?.claimableExternalRewards.length,
   )
 
+  const boost =
+    gauge.gaugeBalance != null &&
+    !gauge.gaugeBalance.isZero() &&
+    gauge.workingBalances != null &&
+    !gauge.workingBalances.isZero()
```
Comment (ug02fast) [inline src/hooks/useUserGauge.ts:64]:
```
gauge?.gaugeBalance?.gt(Zero) &&
gauge?.workingBalances?.gt(Zero)
```
this should accomplish similar if you wanted to simplify

**Review by ug02fast — APPROVED** [review]

File: src/hooks/useUserGauge.ts (hunk 1 above)
Comment (ug02fast) [reply src/hooks/useUserGauge.ts:64]: might just be 
```
gauge.gaugeBalance?.gt(Zero) &&
gauge.workingBalances?.gt(Zero)
```

Comment (nguyenaiden) [reply src/hooks/useUserGauge.ts]: Good call. Will use it now.

File: src/hooks/useUserGauge.ts [hunk 2]
```diff
@@ -54,6 +57,18 @@ export default function useUserGauge(gaugeAddress?: string): UserGauge | null {
     userGaugeRewards?.claimableExternalRewards.length,
   )
 
+  const boost =
+    gauge.gaugeBalance?.gt(Zero) && gauge.workingBalances?.gt(Zero)
+      ? `${formatUnits(
+          gauge.workingBalances
+            .mul(BN_1E18)
+            .div(gauge.gaugeBalance)
```
Comment (hammeiam) [inline src/hooks/useUserGauge.ts:65]: Careful when dividing by unknown values - they may be 0 and BigNumber will throw

**Review by hammeiam — APPROVED** [review]
Looks good besides one gotcha, approving to unblock

File: src/hooks/useUserGauge.ts (hunk 2 above)
Comment (nguyenaiden) [reply src/hooks/useUserGauge.ts:65]: we have a null and non Zero check above.

**Pull request merged**